src/Panel/RoutesPanel.php

```diff
@@ -40,16 +40,14 @@ public function summary(): string
     /**
      * Data collection callback.
      *
-     * @param \Cake\Event\EventInterface $event The shutdown event.
+     * @param \Cake\Event\EventInterface<\Cake\Controller\Controller> $event The shutdown event.
      * @return void
      */
     public function shutdown(EventInterface $event): void
     {
-        /** @var \Cake\Controller\Controller|null $controller */
         $controller = $event->getSubject();
```

Comment:
This line either must return the controller object or throws an exception, if no subject is present. Therefore the whole nullable checks are not needed anymore.